pkgs/tools/networking/dnschef/default.nix

```
@@ -0,0 +1,27 @@
+{python3Packages, buildPythonApplication, fetchFromGitHub, dnslib, lib}:
+
+buildPythonApplication rec {
+  pname = "dnschef";
+  version = "0.4";
+
+  src = fetchFromGitHub {
+    owner = "iphelix";
+    repo = "dnschef";
+    rev = "a395411ae1f5c262d0b80d06a45a445f696f3243";
+    sha256 = "0ll3hw6w5zhzyqc2p3c9443gcp12sx6ddybg5rjpl01dh3svrk1q";
+  };
+
+  format = "other";
+  installPhase = ''
+    install -D ./dnschef.py $out/bin/dnschef
+  '';
+
+  propagatedBuildInputs = [ python3Packages.dnslib ];
```

```suggestion
  propagatedBuildInputs = [ dnslib ];
```

`python3Packages.callPackage` can provide this directly.

Note that without `python3Packages.callPackage`, `dnslib` produces an error because it is not in Nixpkgs' global top level, but you didn't use it in the function body so laziness prevented that error. (Although I suspect you may have encountered it while iterating?)